libp2p/Host.cpp

```diff
@@ -688,12 +555,18 @@ void Host::doWork()
 		m_ioService.run();
 }
 
-void Host::pingAll()
+void Host::keepAlivePeers()
 {
-	RecursiveGuard l(x_peers);
-	for (auto& i: m_peers)
-		if (auto j = i.second.lock())
-			j->ping();
+	RecursiveGuard l(x_sessions);
+	for (auto p: m_sessions)
+		if (auto pp = p.second.lock())
+		{
+			if (chrono::steady_clock::now() - pp->m_lastReceived >= chrono::seconds(60))
```

Comment:
The ping interval (30s) and this number (60s) should be defined close to each other.